suites/src/main/java/com/hedera/block/suites/persistence/positive/PositiveDataPersistenceTests.java

```diff
@@ -70,8 +66,8 @@ public void verifyBlockDataSavedInCorrectDirectory() throws InterruptedException
         String savedBlocksFolderBefore = getContainerCommandResult(GET_BLOCKS_COMMAND);
         int savedBlocksCountBefore = getSavedBlocksCount(savedBlocksFolderBefore);
 
-        BlockStreamSimulatorApp blockStreamSimulatorApp = createBlockSimulator();
-        startSimulatorThread(blockStreamSimulatorApp);
+        blockStreamSimulatorApp = createBlockSimulator();
+        simulatorThread = startSimulatorInThread(blockStreamSimulatorApp);
         Thread.sleep(5000);
```

Comment:
I do not think it is a good idea to `sleep` in tests, this makes them fragile and potentially flaky based on the logic they test. Today they may pass, but tomorrow the sleep time might not be enough. Another downside is that we may `sleep` for a lot longer than needed thus slowing down the tests if we `sleep` a lot in a lot of places. If the sleep here serves only the purpose to wait for the `simulatorThread` `Future` (which I thing should also be renamed to `simulatorFuture` to better represent the object, since the `Future` is not the actual thread), then there are better ways to wait for the future to complete and to not make the test fragile.

Consider this comment for all similar places.

Reply:
Usually I would agree, and you are correct that this pattern is not good. However there are two reasons for the use of sleep here: 
- This was the easiest way, without any overengineering to achive the needed results and test the behaviour I want to test. Other way would be too add pooling mechanism, but wasn't in the scope of the tasks.
- Sleep here is a benefit and needed in our test cases. Here is why: We want to start and stream data to the block-node, and becuase block-node performance is such an integral part of the application, we want to make sure that this data was received and procesed very fast. Therefore waiting a little (5seconds, but we can lower it in the future) and checking, whether the data that we've send was processed will tell us, if block-node is working is expected. And if this fails for some reason, it will signal us that there is something wrong with the performance. Therefore in this case, if this test and the other ones that we us similar sleep fails, it won't be because it's flaky, but because we have a problem with block-node performance, which is good, because we can start investigating.

Reply:
I do agree with you that sleeping is the easiest way to achieve this, for this test even a while not done with an `@Timeout` on the test method for security of the test not going on forever would do, that is IF the Future shall complete soon (in your case 5 secs), otherwise if we only need to start up the Future and let it then go on, that is another case. Yet I do think it is good to take a look at this in an upcoming issue. Would you agree to make an issue and tag it with our `@todo(issueNumber)` convention?

Reply:
`@Timeout` is not suitable here. Sleep is intentional here, this future, in our case simulator can go on forever. We just need it run a little and then check if it's done it's job.